kube-runtime/src/utils/backoff_reset_timer.rs

```diff
@@ -1,86 +1,95 @@
 use std::time::{Duration, Instant};
 
-use backoff::{backoff::Backoff, Clock, SystemClock};
+use backon::BackoffBuilder;
 
-/// A [`Backoff`] wrapper that resets after a fixed duration has elapsed.
-pub struct ResetTimerBackoff<B, C = SystemClock> {
-    backoff: B,
-    clock: C,
-    last_backoff: Option<Instant>,
+use super::{ResettableBackoff, ResettableBackoffWrapper};
+
+#[cfg(doc)] use backon::Backoff;
+
+/// Builder for [`ResetTimerBackoff`].
+#[derive(Debug, Clone)]
+pub struct ResetTimerBackoffBuilder<B> {
```

Comment:
I do see the other PR is sidestepping this by adding a trait instead. https://github.com/kube-rs/kube/pull/1653/files#diff-32fcc57da3d0b322998a85b05a1a82dae3f76df733b8a878efa1c97695af386eL3-R13

not sure if this is better or not, but it's possibly less abstraction overhead?

Reply:
That PR requires you to implement its `Backoff` trait manually for each use case, which adds a lot of (code) overhead, especially for downstream users that would be subject to the orphaning rules.

Reply:
hopefully this is how docs.rs builds docs. i tried to find some info on this, but not much on [their docs](https://docs.rs/about/builds). it is possible to [inject rustdoc args](https://docs.rs/about/metadata) though, so should probably be fine

Reply:
True, but are there realistically that many people that would need to tweak backoff behavior? Do we need to optimise for those people? We don't have examples/ to show people how to do it (which to me is indicative that this is not super high value). I can only think of people __trying__ to use it as a half-baked solution to extremely tight consistency guarantees (by hammering the api-server more during network stress) - and that feels not like something we want to encourage. Maybe I am missing something?

The downside to this is complexity of our side; 4 types that we expose from utils/mod (with relations that are hard to decipher), a generic type that sneaks its way into type signatures (in an already complicated controller/mod).

Reply:
> True, but are there realistically that many people that would need to tweak backoff behavior?

It just feels weird that it becomes.. almost composable.

> a generic type that sneaks its way into type signatures (in an already complicated controller/mod).

We should be able to hide that by integrating it into the implementation of `StreamBackoff`.

Reply:
> We should be able to hide that by integrating it into the implementation of StreamBackoff.

Ah right no, since `BackoffBuilder` doesn't make sense as a trait object, so that complicates the type of `Controller`.